File: basic/src/main/java/com/yuanshenbin/basic/log/ILog.java
package com.yuanshenbin.basic.log;


import com.yuanshenbin.basic.config.BasicOptions;

public class ILog {
    private static LogAbstract instance;

    public static LogAbstract initialize(LogAbstract logAbstract) {
        if (instance == null) {
            instance = logAbstract;
        }
        return instance;
    }

    
    public static void i(String tag, String msg) {
        if (instance!=null&& BasicOptions.getInstance().isDebug()){
            instance.i(tag, msg);
        }
    }


    public static void i(String msg) {
        if (instance!=null&& BasicOptions.getInstance().isDebug()){
            instance.i(msg);
        }
    }


    public static void e(String tag, String msg) {
        if (instance!=null&& BasicOptions.getInstance().isDebug()){
            instance.e(tag, msg);
        }
    }


    public static void e(String msg) {
        if (instance!=null&& BasicOptions.getInstance().isDebug()){
            instance.e(msg);
        }
    }


    public static void v(String tag, String msg) {
        if (instance!=null&& BasicOptions.getInstance().isDebug()){
            instance.v(tag, msg);
        }
    }


    public static void d(String msg) {
        if (instance!=null&& BasicOptions.getInstance().isDebug()){
            instance.d(msg);
        }
    }


    public static void d(String tag, String msg) {
        if (instance!=null){
            instance.d(tag, msg);
        }
    }


    public static void w(String msg) {
        if (instance!=null){
            instance.w(msg);
        }
    }


    public static void w(String tag, String msg) {
        if (instance!=null){
            instance.w(tag, msg);
        }
    }
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/log/ILog.kt
package com.yuanshenbin.basic.log

import com.yuanshenbin.basic.config.BasicOptions


/**
 * author : yuanshenbin
 * time   : 2023/3/9
 * desc   :
 */
object ILog {
    private var instance: LogAbstract? = null

    @JvmStatic
    fun initialize(abstract: LogAbstract?): LogAbstract? {
        if (instance == null) {
            instance = abstract
        }
        return instance
    }

    @JvmStatic
    fun i(tag: String?, msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.i(tag, msg)
        }
    }

    @JvmStatic
    fun i(msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.i(msg)
        }
    }

    @JvmStatic
    fun e(tag: String?, msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.e(tag, msg)
        }
    }

    @JvmStatic
    fun e(msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.e(msg)
        }
    }

    @JvmStatic
    fun v(tag: String?, msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.v(tag, msg)
        }
    }

    @JvmStatic
    fun d(msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.d(msg)
        }
    }

    @JvmStatic
    fun d(tag: String?, msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.d(tag, msg)
        }
    }

    @JvmStatic
    fun w(msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.w(msg)
        }
    }

    @JvmStatic
    fun w(tag: String?, msg: String?) {
        if (BasicOptions.instance.isDebug()!!) {
            instance?.w(tag, msg)
        }
    }
}